public/firebase-messaging-sw.js

```diff
@@ -0,0 +1,21 @@
+importScripts('https://www.gstatic.com/firebasejs/9.6.3/firebase-app.js');
+importScripts('https://www.gstatic.com/firebasejs/9.6.3/firebase-messaging.js');
+importScripts('firebase-config.js');
+
+if (firebaseConfig) {
+  firebase.initializeApp(firebaseConfig);
+  console.log(`${new Date().toJSON()}  [firebase-messaging-sw] Initialized messaging`);
+
+  firebase.messaging().setBackgroundMessageHandler(payload => {
+    console.log(`${new Date().toJSON()}  [firebase-messaging-sw] Received background message`, payload);
+    const notificationTitle = payload.data.twi_message_type;
```

Comment:
here we should use static method Client.parsePushNotification(<pushdata>) and then access it through returned object of type PushNotification